fiboa_cli/publish.py

```diff
@@ -0,0 +1,378 @@
+
+    @staticmethod
+    def get_cli_args():
+        return {
+            **ConvertData.get_cli_args(),
+            "target": VECOREL_TARGET(folder=True),
+            "generate_meta": click.option(
+                "--generate-meta",
+                "-gm",
+                is_flag=True,
+                type=click.BOOL,
+                help="Generate README.txt and LICENSE.txt for the dataset if not present.",
+                default=False,
+            ),
+            "yes": click.option(
+                "--yes",
+                "-y",
+                is_flag=True,
+                type=click.BOOL,
+                help="Answer yes to all questions.",
+                default=False,
+            ),
+            "data_url": click.option(
+                "--data-url",
+                type=click.STRING,
+                help="When generating documentation, this is the link to the data.",
+            ),
+            "s3_upload_path": click.option(
+                "--s3-upload-path",
+                type=click.STRING,
+                help="Upload to this path on S3. By default it's the source coop fiboa data repository.",
+            ),
+        }
+
+    @staticmethod
+    def get_cli_callback(cmd):
+        def callback(dataset, *args, **kwargs):
+            return Publish(dataset).run(*args, **kwargs)
+
+        return callback
+
+    def __init__(self, dataset: str, data_url=None, s3_upload_path=None):
+        super().__init__()
+        self.cmd_title = f"Convert {dataset}"
+        self.dataset = dataset
+        self.data_url = data_url or f"{self.url_base}/{self.dataset}"
+        self.s3_upload_path = (
+            s3_upload_path or f"s3://us-west-2.opendata.source.coop/fiboa/data/{self.dataset}/"
+        )
+
+        if ConvertData.converters.is_converter(self.dataset):
+            raise Exception(f"'{self.dataset}' is not a converter")
+        try:
+            self.converter = ConvertData.converters.load(self.dataset)
+        except (ImportError, NameError, OSError, RuntimeError, SyntaxError) as e:
+            raise Exception(f"Converter for '{self.dataset}' not available or faulty: {e}") from e
+
+    def exc(self, cmd):
+        assert os.system(cmd) == 0
+
+    def check_command(self, cmd, name=None):
+        if os.system(f"{cmd} --version") != 0:
+            self.error(f"Missing command {cmd}. Please install {name or cmd}")
+            sys.exit(1)
+
+    def download_data_survey(self, base):
+        data_survey = (
+            os.getenv("FIBOA_DATA_SURVEY")
+            or f"https://raw.githubusercontent.com/fiboa/data-survey/refs/heads/main/data/{base}.md"
+        )
+        response = requests.get(data_survey)
+        if not response.ok:
+            self.warning(
+                f"Missing data survey {base}.md at {data_survey}. Can not auto-generate file"
+            )
+        else:
+            return response.text
+
+    @cache
+    def get_data_survey(self):
```

Comment:
What is this for? Reading projection and homepage? I feel like the data surveys are a little to inconsistent for this, so wondering if important information is in there, whether we can determine that differently (e.g. projection from the data itself?). Isn't the homepage part of the converter? etc.